tools/scripts/pipeline/build/clearly-defined/check-clearly-defined.ps1

```diff
@@ -0,0 +1,188 @@
+# Copyright (c) Microsoft. All rights reserved.
+# Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+<#
+.SYNOPSIS
+Determine if this is a dependabot PR. If it is and the package being updated is not excluded, check
+if ClearlyDefined has information for the current version. Fail the build is it's not there.
+
+.DESCRIPTION
+ClearlyDefined is our source of license information for dependemncies, but it often lags behind
+the latest version of a package. This script exists to detect this scenario and to serve as a
+reminder to request that ClearlyDefined harvest information for the version being updated. It
+also provides a way to exclude packages that we know will never be in ClearlyDefined.
+
+Please keep this script in sync across the following repos:
+  accessibility-insights-web
+  accessibility-insights-windows
+  axe-windows
+
+.PARAMETER PipelineType
+The pipeline environment. This is used to determine which environment variables specify the branch
+name, as well as how to format the output. Supported values are "action", "ado", and "local".
+
+.PARAMETER BranchName
+Overrides the default branch name detection. Optional for "action" and "ado" pipelines, required
+for "local" pipelines.
+
+.EXAMPLE
+./check-clearly-defined.ps1 -PipelineType <string> [-BranchName <string>]
+#>
+
+[CmdletBinding()]
+Param(
+    [Parameter(Mandatory = $true)]
+    [string]$PipelineType,
+    [Parameter(Mandatory = $false)]
+    [string]$BranchName
+)
+
+Set-StrictMode -Version Latest
+$script:ErrorActionPreference = 'Stop'
+
+function GetPipelineType([string]$pipelineType) {
+    $trimmedType = $pipelineType.Trim()
+    switch ($trimmedType) {
+        "action" {
+            return $trimmedType
+        }
+        "ado" {
+            return $trimmedType
+        }
+        "local" {
+            return $trimmedType
+        }
+    }
+
+    Throw "Pipeline type not provided or invalid"
+}
+
+function GetBranchName([string]$pipelineType, [string]$branchName) {
+    $trimmedBranchName = $branchName.Trim()
+    if ($trimmedBranchName.Length -eq 0) {
+        switch ($pipelineType) {
+            "action" {
+                $trimmedBranchName = ($Env:GITHUB_HEAD_REF).Trim()
+            }
+            "ado" {
+                $prBranchName = $Env:SYSTEM_PULLREQUEST_SOURCEBRANCH
+                if ($prBranchName -eq $null) {
+                    $trimmedBranchName = ($Env:BUILD_SOURCEBRANCH).Trim().Replace("refs/heads/","")
+                } else {
+                    $trimmedBranchName = $prBranchName.Trim()
+                }
```

Comment:
This supports for PR and non-PR builds. Found this one the hard way. We'll eventually want to backport this to web